plugins/launcher_service_plugin/launcher_service_plugin.cpp

```diff
@@ -0,0 +1,1072 @@
+                        ("available_keys", std::move(pub_keys));
+         return call(cluster_id, node_id, "/v1/chain/get_required_keys", get_arg);
+      }
+
+      fc::optional<abi_serializer> abi_serializer_resolver(int cluster_id, int node_id, const name& account) {
+         auto &abi_cache = _running_clusters[cluster_id].nodes[node_id].abi_cache;
+         if (fc::time_point::now() >= _running_clusters[cluster_id].nodes[node_id].abi_cache_time + fc::milliseconds(500)) {
+            _running_clusters[cluster_id].nodes[node_id].abi_cache_time = fc::time_point::now();
+            abi_cache.clear();
+         }
+         auto it = abi_cache.find( account );
+         if ( it == abi_cache.end() ) {
+            auto result = call(cluster_id, node_id, "/v1/chain/get_abi", fc::mutable_variant_object("account_name", account));
+            auto abi_results = result.as<get_abi_results>();
+            fc::optional<abi_serializer> abis;
+            if( abi_results.abi.valid() ) {
+               abis.emplace( *abi_results.abi, _config.abi_serializer_max_time );
+            }
+            abi_cache.emplace( account, abis );
+            return abis;
+         }
+         return it->second;
+      };
+
+      bytes action_variant_to_bin(int cluster_id, int node_id, const name& account, const name& action,
+                                  const fc::variant& action_args_var) {
+         auto abis = abi_serializer_resolver(cluster_id, node_id, account);
+         if (!abis.valid()) {
+            _running_clusters[cluster_id].nodes[node_id].abi_cache.clear();
+            FC_ASSERT(false, "No ABI found for ${contract}", ("contract", account));
+         }
+
+         auto action_type = abis->get_action_type( action );
+         if (action_type.empty()) {
+            _running_clusters[cluster_id].nodes[node_id].abi_cache.clear();
+            FC_ASSERT(false, "Unknown action ${action} in contract ${contract}", ("action", action)( "contract", account ));
+         }
+
+         char temp[512 * 1024];
+         fc::datastream<char*> ds(temp, sizeof(temp)-1 );
+         abis->variant_to_binary( action_type, action_args_var, ds, _config.abi_serializer_max_time );
+         std::vector<char> r;
+         r.assign(temp, temp + ds.tellp());
+         return r;
+      }
+
+      fc::variant push_transaction(int cluster_id, int node_id, signed_transaction& trx,
+                                   const std::vector<public_key_type> &sign_keys = std::vector<public_key_type>(),
+                                   packed_transaction::compression_type compression = packed_transaction::compression_type::none) {
+         int port = _running_clusters[cluster_id].nodes[node_id].http_port;
+         auto &info = _running_clusters[cluster_id].nodes[node_id].get_info_cache;
+         if (fc::time_point::now() >= _running_clusters[cluster_id].nodes[node_id].get_info_cache_time + fc::milliseconds(500)) {
```

Comment:
Use `config::block_interval_ms` instead of 500 if you intend the cache to be invalidated every block.